src/Altinn.App.Core/Internal/Process/IProcessEngine.cs

```diff
@@ -0,0 +1,29 @@
+using Altinn.App.Core.Models;
+using Altinn.Platform.Storage.Interface.Models;
+
+namespace Altinn.App.Core.Internal.Process
+{
+    /// <summary>
+    /// Process engine interface that defines the Altinn App process engine
+    /// </summary>
+    public interface IProcessEngine
+    {
+        /// <summary>
+        /// Method to start a new process
+        /// </summary>
+        Task<ProcessChangeResult> StartProcess(ProcessStartRequest processStartRequest);
+
+        /// <summary>
+        /// Method to move process to next task/event
+        /// </summary>
+        Task<ProcessChangeResult> Next(ProcessNextRequest request);
+
+        /// <summary>
+        /// Update Instance and rerun instance events
+        /// </summary>
+        /// <param name="startRequest"></param>
+        /// <param name="events"></param>
+        /// <returns></returns>
+        Task<Instance> UpdateInstanceAndRerunEvents(ProcessStartRequest startRequest, List<InstanceEvent>? events);
```

Comment:
This seems a bit misplaced compared to the other two and the name causes me to think of side effects and why we need to re-run events.

Reply:
Added an issue for tracking this: #240